lib/src/data/datasource/local/DAOs/firestore_database.dart

```diff
@@ -19,6 +19,53 @@ class FirestoreDatabase {
     batch.commit();
   }
 
+  Future<List<Map<String, dynamic>>> getFavorites() async {
+    final List<Map<String, dynamic>> favorites = [];
+    try {
+      final QuerySnapshot snapshot = await _firestore
+          .collection('cards')
+          .where('favorite', isEqualTo: true)
+          .get();
+
+      favorites.addAll(
+        snapshot.docs.toList().map((e) => e.data()).toList(),
+      );
+    } catch (e) {
+      throw Exception(e);
+    }
+    return favorites;
+  }
+
+  Future<bool> isFavorite(String cardId) async {
+    bool exists = false;
+
+    try {
+      final DocumentSnapshot snapshot =
+          await _firestore.collection('cards').doc(cardId).get();
+      exists = snapshot.data()?['favorite'] as bool? ?? false;
+    } catch (_) {}
+    return exists;
+  }
+
+  Future<bool> setFavorite(String cardId) async {
+    final DocumentReference reference =
+        _firestore.collection('cards').doc(cardId);
+    await reference.update({'favorite': true});
+    return true;
+  }
+
+  Future<bool> removeFavorite(String cardId) async {
+    try {
+      await _firestore
+          .collection('cards')
+          .doc(cardId)
+          .update({'favorite': FieldValue.delete()});
```

Comment:
add collection to constants and favourite field to

Reply:
Done